interop/src/main.rs

```diff
@@ -176,6 +182,50 @@ fn run(log: Logger, options: Opt) -> Result<()> {
         }
     }
 
+    let mut h3 = false;
+    let result = runtime.block_on(
+        endpoint
+            .connect_with(&client_config, &remote, &options.host)?
+            .map_err(|e| format_err!("failed to connect: {}", e))
+            .and_then(|conn| {
+                assert!(state.lock().unwrap().saw_cert);
+                handshake = true;
+                let conn = conn.connection;
+                let control_stream = conn.open_uni();
+                let control_fut = control_stream
+                    .map_err(|e| format_err!("failed to open control stream: {}", e))
+                    .and_then(move |stream| {
+                        let mut buf = BytesMut::new();
+                        buf.put(0x43u8);
+                        HttpFrame::Settings(SettingsFrame {
+                            max_header_list_size: 4096,
+                            num_placeholders: 0,
+                        })
+                        .encode(&mut buf);
+                        tokio::io::write_all(stream, buf)
+                            .map_err(|e| format_err!("failed to send Settings frame: {}", e))
+                            .and_then(move |(_, _)| futures::future::ok(()))
+                    });
+
+                let req_stream = conn.open_bi();
+                let req_fut = req_stream
+                    .map_err(|e| format_err!("failed to open request stream: {}", e))
+                    .and_then(|req_stream| h3_get(req_stream))
+                    .and_then(move |data| {
+                        println!(
+                            "read {} bytes: \n\n{}\n\n closing",
+                            data.len(),
+                            String::from_utf8_lossy(&data)
+                        );
+                        conn.close(0, b"done").map_err(|_| unreachable!())
+                    });
+                control_fut.and_then(|_| req_fut).map(|_| h3 = true)
```

Comment:
You can make these concurrent by replacing `and_then` with `join`

Reply:
Thank you, it's actually serially executed on purpose. Not making any request if the control stream fails.